apps/vaporgui/PMultiVarSelector.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "PWidget.h"
+
+class QListWidget;
+class QListWidgetItem;
+
+//! \class PMultiVarSelector
+//! \brief Allows the selection of multiple variables. To add a title, use a PLabel.
+//! \author Stas Jaroszynski
+//! By default this shows the vars with the same dimension as the renderer. This can be changed
+//! with DisplayVars(Mode)
+
+class PMultiVarSelector : public PWidget {
+public:
+    enum Mode { Auto, D2, D3, Both };
+    
+    PMultiVarSelector(std::string tag);
+    PMultiVarSelector *DisplayVars(Mode mode);
+    
+protected:
+    void updateGUI() const override;
+    QSize minimumSizeHint() const override;
+    bool requireDataMgr() const override { return true; }
+    
+private:
+    QListWidget *_listWidget;
+    Mode _mode = Auto;
+    
+    QListWidgetItem *addVarToList(const std::string &var) const;
+    void itemChanged(QListWidgetItem*);
```

Comment:
The convention says private class method names start from an underscore. 

Should it be enforced? I don't have a strong opinion.

Reply:
You are correct. I have been writing all of my code with pascal case for public and camel case for private/protected functions and _camel for private/protected vars. I find it is useful to have a distinction between private variables and functions. I think it is a minor difference to someone reading the code, especially since my convention shares the same rules in its public interface. Much of the legacy code follows no convention at all so it is not a black sheep in the codebase.

Reply:
@StasJ please follow the convention we've adopted.

Reply:
Done. Just FYI, @clyne @shaomeng both of your last PRs also violate the Vapor coding convention so we may want to look into whether we want to revise it.